### DISCUSSION TITLE
part-ii/2017-22/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#373] [Question comments]
Part II 2017 Automata and Formal Languages: Paper 1, Section I, $4 \mathrm{H}$ 

https://questions.tripos.org/part-ii/2017-22/

**Comment by artie2000** [comment]
<details>
<summary>(b)</summary>

The CFL is $S\to SS\mid S+S\mid S*\mid (S)\mid a_1\mid\dots\mid a_n\mid `\varepsilon'\mid`\emptyset'$.

To show it's not regular, apply pumping lemma to $(^ka_1)^k$.

</details>
